modules/apps/document-library/document-library-preview-document/src/main/resources/META-INF/resources/preview/js/DocumentPreviewer.es.js

```diff
@@ -95,11 +95,19 @@ const DocumentPreviewer = ({baseImageURL, initialPage, totalPages}) => {
 		}, 100);
 	}
 
+	const createImageURL = (page) => {
+		const pos = baseImageURL.search('previewFileIndex=') + 17;
+		const imageURL =
+			baseImageURL.slice(0, pos) + page + baseImageURL.slice(pos);
+
+		return imageURL;
+	};
+
```

Comment:
We can use `Liferay.Util.addParams` instead of manipulating directly the URL string.

I think something like this must work:

```diff
-	const createImageURL = (page) => {
-		const pos = baseImageURL.search('previewFileIndex=') + 17;
-		const imageURL =
-			baseImageURL.slice(0, pos) + page + baseImageURL.slice(pos);
-
-		return imageURL;
- 	};
+ 	const createImageURL = (page) => Liferay.Util.addParams(`previewFileIndex=${page}`, baseImageURL);
```